client/state/geolocation/actions.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * External Dependencies
+ */
+import superagent from 'superagent';
+
+/**
+ * Internal Dependencies
+ */
+import {
+	GEOLOCATION_FETCH,
+	GEOLOCATION_FETCH_COMPLETED,
+	GEOLOCATION_FETCH_FAILED
+} from 'state/action-types';
+
+export function fetchGeolocation( dispatch ) {
+	return () => {
```

Comment:
This should be:
```
export function fetchGeolocation() {
    return ( dispatch ) => {
```

I really suggest adding unit tests for any new actions, reducer, and selectors you add.